commit a3c1e3b732b3708a80e4035b9d845f3f7c7dd0c9
Author: Marc Kleine-Budde <mkl@blackshift.org>
Date:   Wed Feb 8 20:24:29 2012 +0100

    mtd: ixp4xx: oops in ixp4xx_flash_probe
    
    In commit "c797533 mtd: abstract last MTD partition parser argument" the
    third argument of "mtd_device_parse_register()" changed from start address
    of the MTD device to a pointer to a struct.
    
    The "ixp4xx_flash_probe()" function was not converted properly, causing
    an oops during boot.
    
    This patch fixes the problem by filling the needed information into a
    "struct mtd_part_parser_data" and passing it to
    "mtd_device_parse_register()".
    
    Signed-off-by: Marc Kleine-Budde <mkl@blackshift.org>
    Signed-off-by: Artem Bityutskiy <artem.bityutskiy@linux.intel.com>
    Cc: stable@kernel.org [3.2+]
    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>

--- a/drivers/mtd/maps/ixp4xx.c
+++ b/drivers/mtd/maps/ixp4xx.c
@@ -182,6 +182,9 @@ static int ixp4xx_flash_probe(struct pla
 {
 	struct flash_platform_data *plat = dev->dev.platform_data;
 	struct ixp4xx_flash_info *info;
+	struct mtd_part_parser_data ppdata = {
+		.origin = dev->resource->start,
+	};
 	int err = -1;
 
 	if (!plat)
@@ -247,7 +250,7 @@ static int ixp4xx_flash_probe(struct pla
 	/* Use the fast version */
 	info->map.write = ixp4xx_write16;
 
-	err = mtd_device_parse_register(info->mtd, probes, dev->resource->start,
+	err = mtd_device_parse_register(info->mtd, probes, &ppdata,
 			plat->parts, plat->nr_parts);
 	if (err) {
 		printk(KERN_ERR "Could not parse partitions\n");
